backend/x86_gas.ml

```diff
@@ -222,6 +222,154 @@ let print_instr b = function
+  | PADDW (arg1, arg2) -> i2 b "paddw" arg1 arg2
+  | PADDD (arg1, arg2) -> i2 b "paddd" arg1 arg2
+  | PADDQ (arg1, arg2) -> i2 b "paddq" arg1 arg2
+  | ADDPD (arg1, arg2) -> i2 b "addpd" arg1 arg2
+  | PADDSB (arg1, arg2) -> i2 b "paddsb" arg1 arg2
+  | PADDSW (arg1, arg2) -> i2 b "paddsw" arg1 arg2
+  | PADDUSB (arg1, arg2) -> i2 b "paddusb" arg1 arg2
+  | PADDUSW (arg1, arg2) -> i2 b "paddusw" arg1 arg2
+  | PSUBB (arg1, arg2) -> i2 b "psubb" arg1 arg2
+  | PSUBW (arg1, arg2) -> i2 b "psubw" arg1 arg2
+  | PSUBD (arg1, arg2) -> i2 b "psubd" arg1 arg2
+  | PSUBQ (arg1, arg2) -> i2 b "psubq" arg1 arg2
+  | SUBPD (arg1, arg2) -> i2 b "subpd" arg1 arg2
+  | PSUBSB (arg1, arg2) -> i2 b "psubsb" arg1 arg2
+  | PSUBSW (arg1, arg2) -> i2 b "psubsw" arg1 arg2
+  | PSUBUSB (arg1, arg2) -> i2 b "psubusb" arg1 arg2
+  | PSUBUSW (arg1, arg2) -> i2 b "psubusw" arg1 arg2
+  | PMAXUB (arg1, arg2) -> i2 b "pmaxub" arg1 arg2
+  | PMAXSW (arg1, arg2) -> i2 b "pmaxsw" arg1 arg2
+  | MAXPD (arg1, arg2) -> i2 b "maxpd" arg1 arg2
+  | PMINUB (arg1, arg2) -> i2 b "pminub" arg1 arg2
+  | PMINSW (arg1, arg2) -> i2 b "pminsw" arg1 arg2
+  | MINPD (arg1, arg2) -> i2 b "minpd" arg1 arg2
+  | MULPD (arg1, arg2) -> i2 b "mulpd" arg1 arg2
+  | DIVPD (arg1, arg2) -> i2 b "divpd" arg1 arg2
+  | PAND (arg1, arg2) -> i2 b "pand" arg1 arg2
+  | PANDNOT (arg1, arg2) -> i2 b "pandn" arg1 arg2
+  | POR (arg1, arg2) -> i2 b "por" arg1 arg2
+  | PXOR (arg1, arg2) -> i2 b "pxor" arg1 arg2
+  | PMOVMSKB (arg1, arg2) -> i2 b "pmovmskb" arg1 arg2
+  | MOVMSKPD (arg1, arg2) -> i2 b "movmskpd" arg1 arg2
+  | PSLLDQ (bytes, arg1) -> i2 b "pslldq" (Imm (Int64.of_int bytes)) arg1
+  | PSRLDQ (bytes, arg1) -> i2 b "psrldq" (Imm (Int64.of_int bytes)) arg1
+  | PCMPEQB (arg1, arg2) -> i2 b "pcmpeqb" arg1 arg2
+  | PCMPEQW (arg1, arg2) -> i2 b "pcmpeqw" arg1 arg2
+  | PCMPEQD (arg1, arg2) -> i2 b "pcmpeqd" arg1 arg2
+  | PCMPGTB (arg1, arg2) -> i2 b "pcmpgtb" arg1 arg2
+  | PCMPGTW (arg1, arg2) -> i2 b "pcmpgtw" arg1 arg2
+  | PCMPGTD (arg1, arg2) -> i2 b "pcmpgtd" arg1 arg2
+  | CMPPD (cmp, arg1, arg2) -> i2 b ("cmp" ^ string_of_float_condition cmp ^ "pd") arg1 arg2
+  | CVTDQ2PD (arg1, arg2) -> i2 b "cvtdq2pd" arg1 arg2
+  | CVTDQ2PS (arg1, arg2) -> i2 b "cvtdq2ps" arg1 arg2
+  | CVTPD2DQ (arg1, arg2) -> i2 b "cvtpd2dq" arg1 arg2
+  | CVTPD2PS (arg1, arg2) -> i2 b "cvtpd2ps" arg1 arg2
+  | CVTPS2DQ (arg1, arg2) -> i2 b "cvtps2dq" arg1 arg2
+  | CVTPS2PD (arg1, arg2) -> i2 b "cvtps2pd" arg1 arg2
+  | PSLLW (arg1, arg2) -> i2 b "psllw" arg1 arg2
+  | PSLLD (arg1, arg2) -> i2 b "pslld" arg1 arg2
+  | PSLLQ (arg1, arg2) -> i2 b "psllq" arg1 arg2
+  | PSRLW (arg1, arg2) -> i2 b "psrlw" arg1 arg2
+  | PSRLD (arg1, arg2) -> i2 b "psrld" arg1 arg2
+  | PSRLQ (arg1, arg2) -> i2 b "psrlq" arg1 arg2
+  | PSRAW (arg1, arg2) -> i2 b "psraw" arg1 arg2
+  | PSRAD (arg1, arg2) -> i2 b "psrad" arg1 arg2
+  | PSLLWI (bits, arg1) -> i2 b "psllw" (Imm (Int64.of_int bits)) arg1
```

Comment:
I think `(Imm (Int64.of_int bits))` should be done in `Emit`. Same for other instructions.

Reply:
Updated